karpenter-operator/controllers/karpenter/karpenter_controller.go

```diff
@@ -138,6 +147,36 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		}
 		return ctrl.Result{}, err
 	}
+
+	// Setup for ControlPlaneContext and the Karpenter control plane v2 component.
+	pullSecret := common.PullSecret(hcp.Namespace)
+	if err := r.ManagementClient.Get(ctx, client.ObjectKeyFromObject(pullSecret), pullSecret); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get pull secret: %w", err)
+	}
+
+	releaseClientProvider := &releaseinfo.RegistryClientProvider{}
+	releaseImage, err := releaseClientProvider.Lookup(ctx, util.HCPControlPlaneReleaseImage(hcp), pullSecret.Data[corev1.DockerConfigJsonKey])
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to lookup release image: %w", err)
+	}
+	imageProvider := imageprovider.New(releaseImage)
+	imageProvider.ComponentImages()["token-minter"] = r.ControlPlaneOperatorImage
+	imageProvider.ComponentImages()[util.AvailabilityProberImageName] = r.ControlPlaneOperatorImage
+
+	cpContext := controlplanecomponent.ControlPlaneContext{
+		Context:              ctx,
+		Client:               r.ManagementClient,
+		ApplyProvider:        upsert.NewApplyProvider(false),
+		HCP:                  hcp,
+		ReleaseImageProvider: imageProvider,
+		OmitOwnerReference:   true,
```

Comment:
```suggestion
```
why is this true? the current code seems to be adding the HCP ownerRef to karpenter resources. 
related to my other comment https://github.com/openshift/hypershift/pull/6187#discussion_r2109013495